spanner/client_test.go

```diff
@@ -2239,3 +2239,169 @@ func TestClient_DoForEachRow_ShouldEndSpanWithQueryError(t *testing.T) {
 		t.Errorf("Span status mismatch\nGot: %v\nWant: %v", s.Code, codes.InvalidArgument)
 	}
 }
+
+func TestClient_ReadOnlyTransaction_Tag(t *testing.T) {
```

Comment:
Better to have separate tests for SingleUse read/query, PartitionedDML and BatchDML requests.